include/node.tcc

```diff
@@ -539,3 +539,19 @@ void mpm::Node<Tdim, Tdof, Tnphases>::update_property(
   property_handle_->update_property(property, prop_id_, mat_id, property_value,
                                     nprops);
 }
+
+//! Compute multimaterial change in momentum
+template <unsigned Tdim, unsigned Tdof, unsigned Tnphases>
+void mpm::Node<Tdim, Tdof,
+               Tnphases>::compute_multimaterial_change_in_momentum() {
+  // iterate over all materials in the material_ids set and update the change in
+  // momentum
+  std::lock_guard<std::mutex> guard(node_mutex_);
+  for (auto mat_itr = material_ids_.begin(); mat_itr != material_ids_.end();
+       ++mat_itr)
+    property_handle_->update_property(
+        "change_in_momenta", prop_id_, *mat_itr,
+        velocity_ * property_handle_->property("masses", prop_id_, *mat_itr) -
```

Comment:
Could you take these calculations out separately add a const store it in a variable and use the variables here in the function call, otherwise it's confusing. 

Please use `mitr` instead of `mat_itr` as it is easier to read.